pkg/controllermanager/controller/shoot/shoot_hibernation_control.go

```diff
@@ -218,37 +123,41 @@ func (r *shootHibernationReconciler) Reconcile(ctx context.Context, request reco
 	logger.Info("[SHOOT HIBERNATION]")
 
 	if shoot.DeletionTimestamp != nil {
-		r.deleteShootCron(logger, key)
 		return reconcile.Result{}, nil
 	}
 
-	return reconcile.Result{}, r.createOrUpdateShootCron(ctx, logger, key, shoot)
+	return reconcile.Result{}, nil
 }
 
-func (r *shootHibernationReconciler) deleteShootCron(logger logrus.FieldLogger, key string) {
-	if sched, ok := r.hibernationScheduleRegistry.Load(key); ok {
-		sched.Stop()
-		logger.Debugf("Stopped cron")
-	}
+// GroupHibernationSchedulesByLocation groups the given HibernationSchedules by their Location.
+// If the Location of a HibernationSchedule is `nil`, it is defaulted to UTC.
+func GroupHibernationSchedulesByLocation(schedules []gardencorev1beta1.HibernationSchedule) map[string][]gardencorev1beta1.HibernationSchedule {
+	var (
+		locationToSchedules = make(map[string][]gardencorev1beta1.HibernationSchedule)
+	)
 
-	r.hibernationScheduleRegistry.Delete(key)
-	logger.Debugf("Deleted cron")
-}
+	for _, schedule := range schedules {
+		var locationID string
+		if schedule.Location != nil {
+			locationID = *schedule.Location
+		} else {
+			locationID = time.UTC.String()
+		}
 
-func (r *shootHibernationReconciler) createOrUpdateShootCron(ctx context.Context, logger logrus.FieldLogger, key string, shoot *gardencorev1beta1.Shoot) error {
-	r.deleteShootCron(logger, key)
-	if !shootHasHibernationSchedules(shoot) {
-		return nil
+		locationToSchedules[locationID] = append(locationToSchedules[locationID], schedule)
 	}
 
-	schedule, err := ComputeHibernationSchedule(ctx, r.gardenClient, logger, r.recorder, shoot)
-	if err != nil {
-		return err
-	}
+	return locationToSchedules
+}
 
-	schedule.Start()
-	r.hibernationScheduleRegistry.Store(key, schedule)
-	logger.Debugf("Successfully started hibernation schedule")
+func shootHasHibernationSchedules(shoot *gardencorev1beta1.Shoot) bool {
+	return getShootHibernationSchedules(shoot) != nil
+}
```

Comment:
This functions seems somewhat useless. You could just check for 

```golang
len(getShootHibernationSchedules(shoot)) > 0
```

I would vote for dropping it.